app/services/create_my_module.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+class CreateMyModule
+  include Canaid::Helpers::PermissionsHelper
+
+  # config example: {project_id: 5, experiment_name: 'New Experiment', my_module_name: 'New Task'}
+
+  def initialize(user, team, config)
+    @config = config
+    @user = user
+    @team = team
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      @project = load_project
+      @experiment = load_experiment
+      raise ActiveRecord::Rollback unless @project && @experiment && can_manage_experiment?(@user, @experiment)
+
+      my_module_params = {
+        x: 0,
+        y: 0,
+        name: @config[:my_module_name] || I18n.t('create_task_service.default_task_name')
+      }
+
+      my_module = @experiment.my_modules.new(my_module_params)
+      new_pos = my_module.get_new_position
+      my_module.x = new_pos[:x]
+      my_module.y = new_pos[:y]
+
+      if my_module.save
```

Comment:
Just use save! As it is in transaction, in case of failure all changes will be rolled back.